audio_reader.py

```diff
@@ -0,0 +1,102 @@
+import fnmatch
+import os
+import re
+import threading
+
+import librosa
+import numpy as np
+import tensorflow as tf
+
+
+def find_files(directory, pattern='*.wav'):
+    files = []
+    for root, dirnames, filenames in os.walk(directory):
+        for filename in fnmatch.filter(filenames, pattern):
+            files.append(os.path.join(root, filename))
+    return files
+
+
+def load_generic_audio(directory, sample_rate):
+    files = find_files(directory)
+    for f in files:
+        audio, sr = librosa.load(f, sr=sample_rate, mono=True)
+        audio = audio.reshape(-1, 1)
+        yield audio
+
+
+def load_vctk_audio(directory, sample_rate):
+    files = find_files(directory)
+    speaker_re = re.compile(r'p([0-9]+)_([0-9]+)\.wav')
+    for j, f in enumerate(files):
+        audio, sr = librosa.load(f, sr=sample_rate, mono=True)
+        audio = audio.reshape(-1, 1)
+        speaker_id, recording_id = [int(i) for i in speaker_re.findall(f)[0]]
+        yield audio, speaker_id
+
+
+def trim_sample(audio, threshold=0.3):
+    '''Removes silence in the beginning and end of a sample'''
+    energy = librosa.feature.rmse(audio)
+    frames = np.nonzero(energy > threshold)
+    indices = librosa.core.frames_to_samples(frames)[1]
+    return audio[indices[0]:indices[-1]]
+
+
+class AudioReader(object):
```

Comment:
Missing docstrings for the class, constructor (args),  and methods. This can go in a separate PR though.